### PR TITLE
tests: fix key manager flaky tests

### PR DESCRIPTION
**Pull request by skyline75489**
Signed-off-by: Chester Liu <skyline75489@outlook.com>

### What problem does this PR solve?

Problem Summary:

Some of the key_manager tests are flaky on my system:

```text
thread 'manager::tests::test_key_manager_rotate_master_key' panicked at 'assertion failed: `(left == right)`
  left: `2`,
 right: `1`', components/encryption/src/manager/mod.rs:889:9

...

failures:
    manager::tests::test_key_manager_rotate_master_key
    manager::tests::test_key_manager_rotate_master_key_rewrite_failure
    manager::tests::test_master_key_failure_and_succeed
```
### What is changed and how it works?

What's Changed:

I suspect the failure is caused by locking error so I added related locking code.
### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- PR to update `pingcap/tidb-ansible`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

- No release note

### CODE REVIEW

File: components/encryption/src/manager/mod.rs [hunk 1]
```diff
@@ -859,7 +860,7 @@ mod tests {
     // If master_key is the wrong key, fallback to previous_master_key.
     #[test]
     fn test_key_manager_rotate_master_key() {
-        let mut _guard = LOCK_FOR_GAUGE.lock().unwrap();
+        let _guard = LOCK_FOR_GAUGE.lock().unwrap();
```
Comment (skyline75489) [inline components/encryption/src/manager/mod.rs:863]: The `mut` here does not make sense to me.

File: components/encryption/src/manager/mod.rs [hunk 2]
```diff
@@ -812,6 +812,7 @@ mod tests {
 
     #[test]
     fn test_key_manager_encryption_enable_disable() {
+        let _guard = LOCK_FOR_GAUGE.lock().unwrap();
```
Comment (skyline75489) [inline components/encryption/src/manager/mod.rs:815]: Adding this fixes the tests. I'm not sure what else should this be added.

**Comment by tabokie** [comment]
Thanks for contributing.
The `LOCK_FOR_GAUGE` is required not only when tests read gauge value, but also when tests could potentially mutate the gauge. @hunterlxt Mind help triage if any other tests need fixing?

**Review by hunterlxt — APPROVED** [review]
lgtm

**Comment by hunterlxt** [comment]
> Thanks for contributing.
> The `LOCK_FOR_GAUGE` is required not only when tests read gauge value, but also when tests could potentially mutate the gauge. @hunterlxt Mind help triage if any other tests need fixing?

It would be better to add the `LOCK_FOR_GAUGE` into all unit tests because `Dict::open()` could modified the var and the behavior  happens in almost all tests. @skyline75489

**Comment by skyline75489** [comment]
> add the LOCK_FOR_GAUGE into all unit tests because Dict::open() could modified the var and the behavior  happens in almost all tests

This is also my initial thought. Thanks for confirming this. I've added the locking code.

By the way do I need to manually squash the commits into one?

Have a nice weekend 😄

**Comment by tabokie** [comment]
> > add the LOCK_FOR_GAUGE into all unit tests because Dict::open() could modified the var and the behavior  happens in almost all tests
> 
> This is also my initial thought. Thanks for confirming this. I've added the locking code.
> 
> By the way do I need to manually squash the commits into one?
> 
> Have a nice weekend 😄

No need, the bot will squash them for you.

**Review by tabokie — APPROVED** [review]
LG

**Pull request merged**